src/SquidConfig.cc

```diff
@@ -9,6 +9,7 @@
 #include "squid.h"
 #include "SquidConfig.h"
 
+// TODO: Who is zeroing this global during startup? C++ does not do it for us!
```

Comment:
As I understand it there is an edge case of C++ ... because the variable is global static it is default-initialized as if it had a zero-initializer list. In that situation pointers are implicitly set to nullptr, which happens to prevent crashing during the first parse of Foo::TheConfig and such classes. Basic typed members do not get set until explicitly initialized by the first call to free_all() from configFreeMemory().

Reply:
I figured it out. `Config` is a non-local variable with static [storage duration](https://en.cppreference.com/w/cpp/language/storage_duration) (and external linkage). Its zero-initialization is guaranteed during static initialization (i.e. the first initialization stage during startup) by the corresponding C++ [rules](https://en.cppreference.com/w/cpp/language/initialization#Non-local_variables). Its dynamic initialization (i.e. the second initialization stage during startup) is currently broken (the generated default SquidConfig constructor will do nothing; it will not zero pointers, etc.), but the first stage is sufficient to mask the lack of proper dynamic initialization for this specific SquidConfig instance.

Thus, our `Config` is zero-intialized as expected. This branch TODO is addressed, and I will remove it.

P.S. FWIW, when I wrote that TODO, I was confused by the "otherwise, nothing is done" statement in the [default initialization](https://en.cppreference.com/w/cpp/language/default_initialization) rules. That misleading statement applies to _dynamic_ initialization only. For [non-locals](https://en.cppreference.com/w/cpp/language/initialization#Non-local_variables) like `Config`, we still need to take into account _static initialization_ during (i.e. the first startup initialization stage), which saves the day.